src/main/java/org/opensearch/ad/transport/SearchAnomalyResultTransportAction.java

```diff
@@ -161,7 +183,9 @@ protected void doExecute(Task task, SearchRequest request, ActionListener<Search
                 listener.onFailure(e);
             }
         } else {
+            // onlyQueryCustomResultIndex is false in this branch
             // Search only default result index
+            request.indices(ALL_AD_RESULTS_INDEX_PATTERN);
```

Comment:
Since we are now skipping the permission checks in the `if` case will searchHandler handle search each index and seeing if user has permission to read it?

Reply:
Default result index is system index, so only super admin has permission to read it. So we don't need to do permission check, if user have search AD result permission, they should be able to query default AD result index.